fluent-resmgr/src/resource_manager.rs

```diff
@@ -27,58 +27,90 @@ impl ResourceManager {
+        Ok(if let Some(res) = self.resources.get(&path) {
             res
         } else {
-            let string = read_file(&path).unwrap();
+            let string = read_file(&path)?;
             let res = match FluentResource::try_new(string) {
                 Ok(res) => res,
                 Err((res, _err)) => res,
             };
             self.resources.insert(path.to_string(), Box::new(res))
-        }
+        })
     }
 
     pub fn get_bundle(
         &self,
         locales: Vec<LanguageIdentifier>,
         resource_ids: Vec<String>,
-    ) -> FluentBundle<&FluentResource> {
+    ) -> Result<FluentBundle<&FluentResource>, Vec<ResourceManagerError>> {
+        let mut errors: Vec<ResourceManagerError> = vec![];
         let mut bundle = FluentBundle::new(locales.clone());
-        for res_id in &resource_ids {
-            let res = self.get_resource(res_id, &locales[0].to_string());
-            bundle.add_resource(res).unwrap();
+
+        resource_ids.iter().for_each(|res_id| {
+            self.get_resource(res_id, &locales[0].to_string())
+                .map_err(|err| errors.push(err))
+                .and_then(|res| {
+                    bundle.add_resource(res).map_err(|errs| {
+                        errs.into_iter()
+                            .for_each(|err| errors.push(ResourceManagerError::Fluent(err)))
+                    })
+                })
+                .unwrap();
+        });
+
+        if errors.iter().len() > 0 {
+            Err(errors)
+        } else {
+            Ok(bundle)
         }
-        bundle
     }
 
     pub fn get_bundles(
         &self,
         locales: Vec<LanguageIdentifier>,
         resource_ids: Vec<String>,
-    ) -> impl Iterator<Item = FluentBundle<&FluentResource>> {
-        let res_mgr = self;
+    ) -> Result<impl Iterator<Item = FluentBundle<&FluentResource>>, Vec<ResourceManagerError>>
+    {
         let mut ptr = 0;
+        let mut errors: Vec<ResourceManagerError> = vec![];
+        let mut res = vec![];
 
-        iter::from_fn(move || {
-            locales.get(ptr).map(|locale| {
-                ptr += 1;
-                let mut bundle = FluentBundle::new(vec![locale.clone()]);
-                for res_id in &resource_ids {
-                    let res = res_mgr.get_resource(res_id, &locale.to_string());
-                    bundle.add_resource(res).unwrap();
-                }
-                bundle
-            })
-        })
+        while let Some(locale) = locales.get(ptr) {
+            ptr += 1;
+
+            match self.get_bundle(vec![locale.clone()], resource_ids.clone()) {
+                Ok(bundle) => res.push(bundle),
+                Err(mut errs) => errors.append(&mut errs),
+            };
+        }
+
+        if errors.iter().len() > 0 {
+            Err(errors)
+        } else {
+            Ok(res.into_iter())
+        }
     }
 }
 
+#[derive(Error, Debug)]
+pub enum ResourceManagerError {
```

Comment:
Suggestion: This will need some documentation since it's public.

Reply:
Added here https://github.com/projectfluent/fluent-rs/pull/282/commits/5ad427088f39215d19f2892c824f6155598677ef

Reply:
Praise: Thanks for the docs!